tests/test_gui_ca_exp_creator.py

```diff
@@ -0,0 +1,271 @@
+import os
+import tempfile
+from unittest.mock import MagicMock
+
+# import numpy as np
+import pytest
+
+# CA_main,; plot,; preview_CA,; write_to_file_CA,
+from src.GUI.code.GUI_CA_exp_creator import (
+    CA_window_writer,
+    check,
+    config_writer_CA,
+    get_AP_parameters,
+    get_parameters,
+    load_APwindow,
+    load_folder,
+    save_file,
+    time_converter,
+)
+
+# Sample data for testing
+sample_AP_params = [4.798, 1.03, 0, 0.202, 0.003, 9]
+sample_exp_params = {
+    "experiment_file_name": "test_output",
+    "experiment_file_path": tempfile.gettempdir(),
+    "experiment_rest_time": "10",
+    "experiment_step_number": "5",
+    "experiment_voltage": "0.5",
+    "experiment_time": "60",
+}
+
+
+# Mocking the UI class
+class MockUiCA:
```

Comment:
do we want this in the `conftest.py` file as a fixture as we will reuse it in other tests?